src/compiler/checker.ts

```diff
@@ -5612,7 +5698,11 @@ namespace ts {
                 }
             }
 
+            /**
+             * See signatureAssignableTo, signatureAssignableTo
+             */
             function signatureRelatedTo(source: Signature, target: Signature, reportErrors: boolean): Ternary {
+                // TODO (drosen): De-duplicate code between related functions.

```

Comment:
I have a separate branch for that work right now, but I'd like to get it in as part of a separate PR.